### PR TITLE
Feature/disapprove incident

### PR DESCRIPTION
**Pull request by kcjerrell**
# Description

[Rambly explanation part 1](https://www.loom.com/share/1b9292263e994cd687db22982455204b)

This PR allows for changing an incidents status from 'approved' back to 'pending'. It also removes unnecessary/duplicate code and simplifies in AdminDashboard and its subcomponents.

This PR might be more appropriately called refactor/admin-dashboard.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [x] Complete, but not tested (may need new tests)

# How Has This Been Tested?

- [x] `npm test`
- [x] manual testing

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: src/components/AdminDashboard/AdminDashboard.js
Comment (ahmedseragcodes) [inline src/components/AdminDashboard/AdminDashboard.js:33]: I know a lot of this state was being used, so it's possible in your reformatted version this component state is no longer necessary, but I'd love to see all this functionality restored through your reformat in a loom video walkthrough prior to approving for merge.

File: src/components/AdminDashboard/AdminDashboard.js
Comment (ahmedseragcodes) [inline src/components/AdminDashboard/AdminDashboard.js:92]: So this useEffect was to load up the approved list view via that axios request, is this just being done elsewhere now?

File: src/components/AdminDashboard/AdminDashboard.js
Comment (ahmedseragcodes) [inline src/components/AdminDashboard/AdminDashboard.js:88]: Hmmm it looks like getData is taking an argument for Unapproved incidents, but in your new function for fetchIncidents, you're getting Approved incidents. Are these functions doing two different things and do we need them both?

File: src/components/AdminDashboard/AdminDashboard.js
Comment (ahmedseragcodes) [inline src/components/AdminDashboard/AdminDashboard.js:101]: fetchIncidents sets the approved incidents, but this useEffect was setting form responses, do we just not need it to do that anymore because that component state isn't necessary?

File: src/components/AdminDashboard/AdminDashboard.js
Comment (ahmedseragcodes) [inline src/components/AdminDashboard/AdminDashboard.js:120]: I'm assuming this was just moved, but going to leave a comment here to come back to this in-case I don't see this functionality reproduced further down.

File: src/components/AdminDashboard/AdminDashboard.js
Comment (ahmedseragcodes) [inline src/components/AdminDashboard/AdminDashboard.js:128]: same concern here as above, need to see this functionality reproduced

File: src/components/AdminDashboard/AdminDashboard.js
Comment (ahmedseragcodes) [inline src/components/AdminDashboard/AdminDashboard.js:158]: Somewhat confused by why we're removing lines 145-157 here.

File: src/components/AdminDashboard/AdminDashboard.js
Comment (ahmedseragcodes) [inline src/components/AdminDashboard/AdminDashboard.js:148]: Same concern as above with last 4 functions removed, just unaware of why they were removed / how that functionality is reproduced, but awaiting loom walkthrough to see

File: src/components/AdminDashboard/AdminDashboard.js
Comment (ahmedseragcodes) [inline src/components/AdminDashboard/AdminDashboard.js:195]: What happened to this logout functionality?

File: src/components/AdminDashboard/AdminDashboard.js
Comment (ahmedseragcodes) [inline src/components/AdminDashboard/AdminDashboard.js:243]: So without this, the ability to create an incident from any list view is lost, unless you've completely separated that code from every listview. I'll await the loom video to see this functionality re-created.

File: src/components/AdminDashboard/AntTableComponents/AntTable.js
Comment (ahmedseragcodes) [inline src/components/AdminDashboard/AntTableComponents/AntTable.js:133]: From my understanding, this is a custom Ant Table to be utilized throughout the App. Are we sure none of those props that are being deleted are being utilized whereever this table is being utilized? Has functionality been tested where this table is being used to make sure no bugs are created? Just want to make sure with these types of changes.

File: src/components/AdminDashboard/Incidents.js
Comment (ahmedseragcodes) [inline src/components/AdminDashboard/Incidents.js:25]: This is a lot of change, going to have to see the loom video before considering approval.

**Review by ahmedseragcodes — CHANGES_REQUESTED** [review]
I can see a lot of thought and work went into this reformatting, which is great. There's a lot of state gutting and changes in functionality around key components that I don't necessarily see being reproduced elsewhere though,  so I'll have to wait and see the loom video, because there's risk for potential bugs with this large of a change.

File: src/components/AdminDashboard/AdminDashboard.js
Comment (kcjerrell) [reply src/components/AdminDashboard/AdminDashboard.js:33]: Yeah, I realize there's a lot of stuff cut out in here. I'm about to work on a video now.

File: src/components/AdminDashboard/AntTableComponents/AntTable.js
Comment (kcjerrell) [reply src/components/AdminDashboard/AntTableComponents/AntTable.js:133]: I did a search for usages of AntTable, and there are only 2: Incidents.js (which has been cleaned up and simplified) and ApprovedIncidents.js (which can/will be deleted, because Incidents.js can be used now to display any list of incidents you hand it.)

I'll admit I didn't spend a lot of time on AntTable.  But the biggest changes here are the removal of unused props/imports, and the removal of listToUse, which is no longer necessary because it simply renders the incidents you pass into it.

File: src/components/AdminDashboard/Incidents.js
Comment (kcjerrell) [reply src/components/AdminDashboard/Incidents.js:25]: I touch on this in the video. The callbacks and state being passed in is mostly for dealing with the Approve/Reject buttons. Those buttons have been pulled out into their own component, and Incidents.js sole purpose is to show a list of selectable incidents.

Comment (kcjerrell) [reply src/components/AdminDashboard/Incidents.js:25]: Although probably this component is not even necessary: AntTable could be used directly from AdminDashboard since this doesn't actually do anything but pass props a level deeper.

File: src/components/AdminDashboard/AdminDashboard.js
Comment (kcjerrell) [reply src/components/AdminDashboard/AdminDashboard.js:88]: I am not using the original getData() function anymore (because I hate passing around state setters like that). Three new functions have been added to AdminDashboardHelpers for fetching the three incident lists (approved, pending, form-responses);

File: src/components/AdminDashboard/AdminDashboard.js
Comment (kcjerrell) [reply src/components/AdminDashboard/AdminDashboard.js:195]: screenshot

It's in the navbar now.

File: src/components/AdminDashboard/AdminDashboard.js
Comment (kcjerrell) [reply src/components/AdminDashboard/AdminDashboard.js:243]: It's actually still there, git just didn't line it up right. (see line 211 in the updated file).

File: src/components/AdminDashboard/AdminDashboard.js
Comment (kcjerrell) [reply src/components/AdminDashboard/AdminDashboard.js:92]: Yes, originally there were three separate useEffect hooks for pending/approved/form incidents, and you could tell they were written at different times because they were each different. They've been rewritten and combined into a single function, and called from a single useEffect();

File: src/components/AdminDashboard/AdminDashboard.js
Comment (kcjerrell) [reply src/components/AdminDashboard/AdminDashboard.js:101]: The form-response data is now fetched on line 92 (currently), called from the useEffect just below

File: src/components/AdminDashboard/AdminDashboard.js
Comment (kcjerrell) [reply src/components/AdminDashboard/AdminDashboard.js:120]: This was removed because the AntTable component handles pagination entirely on its own.

File: src/components/AdminDashboard/AdminDashboard.js
Comment (kcjerrell) [reply src/components/AdminDashboard/AdminDashboard.js:128]: Same response as above (AntTable handles selection itself, and passes it back up through setSelectedIds).

File: src/components/AdminDashboard/AdminDashboard.js
Comment (kcjerrell) [reply src/components/AdminDashboard/AdminDashboard.js:158]: They're actually still there, lines 121-131.

File: src/components/AdminDashboard/AdminDashboard.js
Comment (kcjerrell) [reply src/components/AdminDashboard/AdminDashboard.js:148]: These are the event handlers for the approve/reject (and then the following confirm/cancel), and are now in the IncidentStatus component.

**Pull request closed without merging**
